homeassistant/components/doorbird/util.py

```diff
@@ -11,9 +11,32 @@ def get_mac_address_from_doorstation_info(doorstation_info):
 
 
 def get_doorstation_by_token(hass, token):
+    """Get doorstation by token."""
+    return _get_doorstation_by_attr(hass, "token", token)
+
+
+def get_doorstation_by_slug(hass, slug):
     """Get doorstation by slug."""
-    for config_entry_id in hass.data[DOMAIN]:
-        doorstation = hass.data[DOMAIN][config_entry_id][DOOR_STATION]
+    return _get_doorstation_by_attr(hass, "slug", slug)
+
+
+def _get_doorstation_by_attr(hass, attr, val):
+    for entry in hass.data[DOMAIN].values():
+        if DOOR_STATION not in entry:
+            continue
 
-        if token == doorstation.token:
+        doorstation = entry[DOOR_STATION]
+
+        if getattr(doorstation, attr) == val:
```

Comment:
Not sure if it applies to these objects, but if `attr` does not exist on `doorstation`, this method will raise. Pass `None` as third argument to avoid raising.

Reply:
Thanks.  It shouldn't be an issue in this case since both uses cases will always exist.